src/createRollup.ts

```diff
@@ -0,0 +1,210 @@
+import { Address, Chain, PrivateKeyAccount, PublicClient, zeroAddress } from 'viem';
+
+import { CoreContracts } from './types/CoreContracts';
+import { createRollupPrepareTransactionRequest } from './createRollupPrepareTransactionRequest';
+import {
+  CreateRollupTransactionReceipt,
+  createRollupPrepareTransactionReceipt,
+} from './createRollupPrepareTransactionReceipt';
+import { createRollupEnoughCustomFeeTokenAllowance } from './createRollupEnoughCustomFeeTokenAllowance';
+import { createRollupPrepareCustomFeeTokenApprovalTransactionRequest } from './createRollupPrepareCustomFeeTokenApprovalTransactionRequest';
+import { getBlockExplorerUrl } from './utils/getters';
+import {
+  CreateRollupTransaction,
+  createRollupPrepareTransaction,
+} from './createRollupPrepareTransaction';
+import { CreateRollupParams } from './types/createRollupTypes';
+
+type PublicClientWithDefinedChain = Omit<PublicClient, 'chain'> & {
+  chain: Chain;
+};
```

Comment:
this could be useful elsewhere, so let's move it to `src/types/ParentChain.ts`
we can add this:

```

export type ParentChainPublicClient = Prettify<
  Omit<PublicClient, 'chain'> & {
    chain: Prettify<Omit<Chain, 'id'> & { id: ParentChainId }>;
  }
>;

export function validateParentChainPublicClient(
  publicClient: PublicClient,
): ParentChainPublicClient {
  const chainId = publicClient.chain?.id;

  if (!isValidParentChainId(chainId)) {
    throw new Error(`Parent chain not supported: ${chainId}`);
  }

  return publicClient as ParentChainPublicClient;
}
```

and later use it as:

```
const validatedPublicClient = validateParentChainPublicClient(publicClient)
```